preview/MsixCore/MsixCoreInstaller/ValidateTargetDeviceFamily.cpp

```diff
@@ -58,33 +45,60 @@ HRESULT ValidateTargetDeviceFamily::ParseTargetDeviceFamilyFromPackage()
-
-    Text<wchar_t> targetDeviceFamilyName;
-    RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"Name", &targetDeviceFamilyName));
-    m_targetDeviceFamilyName = targetDeviceFamilyName.Get();
-
-    Text<wchar_t> minVersion;
-    RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"MinVersion", &minVersion));
-    std::wstring manifestMinVersion = minVersion.Get();
-
-    /// Major version
-    size_t start = 0;
-    size_t end = manifestMinVersion.find_first_of(L'.');
-    m_majorVersion = manifestMinVersion.substr(start, end - start);
-
-    /// Minor version
-    manifestMinVersion.replace(start, end - start + 1, L"");
-    end = manifestMinVersion.find_first_of(L'.');
-    m_minorVersion = manifestMinVersion.substr(start, end - start);
+    while (hc)
+    {
+        ComPtr<IMsixElement> dependencyElement;
+        RETURN_IF_FAILED(dependencyEnum->GetCurrent(&dependencyElement));
+
+        Text<wchar_t> targetDeviceFamilyName;
+        RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"Name", &targetDeviceFamilyName));
+        m_targetDeviceFamilyName = targetDeviceFamilyName.Get();
+
+        Text<wchar_t> minVersion;
+        RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"MinVersion", &minVersion));
+        std::wstring manifestMinVersion = minVersion.Get();
+
+        /// Major version
+        size_t start = 0;
+        size_t end = manifestMinVersion.find_first_of(L'.');
+        m_majorVersion = manifestMinVersion.substr(start, end - start);
+
+        /// Minor version
+        manifestMinVersion.replace(start, end - start + 1, L"");
+        end = manifestMinVersion.find_first_of(L'.');
+        m_minorVersion = manifestMinVersion.substr(start, end - start);
+
+        /// Build number
+        manifestMinVersion.replace(start, end - start + 1, L"");
+        end = manifestMinVersion.find_first_of(L'.');
+        m_buildNumber = manifestMinVersion.substr(start, end - start);
+
+        /// Return if any one of the target device families are compatible with OS
+        if (IsTargetDeviceFamilyNameCompatibleWithOS() && IsManifestVersionCompatibleWithOS())
+        {
+            TraceLoggingWrite(g_MsixTraceLoggingProvider,
+                "Target device family name and manifest min version are compatible with OS",
+                TraceLoggingLevel(WINEVENT_LEVEL_INFO),
+                TraceLoggingValue(m_targetDeviceFamilyName.c_str(), "TargetDeviceFamilyName"),
+                TraceLoggingValue(manifestMinVersion.c_str(), "ManifestMinVersion"));
+            return S_OK;
+        }
+        else
+        {
+            TraceLoggingWrite(g_MsixTraceLoggingProvider,
+                "Target device family name and manifest min version are not compatible with OS",
+                TraceLoggingLevel(WINEVENT_LEVEL_INFO),
+                TraceLoggingValue(m_targetDeviceFamilyName.c_str(), "TargetDeviceFamilyName"),
+                TraceLoggingValue(manifestMinVersion.c_str(), "ManifestMinVersion"));

```

Comment:
>manifestMinVersion [](start = 34, length = 18)

manifestMinVersion gets truncated as you process it, the original string should be minVersion.Get()

Reply:
Updated.